packages/loader/container-loader/src/container.ts

```diff
@@ -1837,7 +1837,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     ) {
         assert(this._context?.disposed !== false, 0x0dd /* "Existing context not disposed" */);
         // If this assert fires, our state tracking is likely not synchronized between COntainer & runtime.
-        if (this._dirtyContainer) {
+        if (!this._dirtyContainer) {
```

Comment:
I'd think the purpose of this check was - log that we are losing user data if we are forcing to reload runtime.
I.e., it's less about first runtime usage (likely reason you are hitting this error and changing condition), but about any further reloads and losing user data because of it.
That said, when I looked at usage of instantiateContext(), sounds like it's used only when we boot container. If so (please validate), then I'd rather remove this check and error telemetry.

Reply:
@vladsud good to know this. I'll double check it. I think I had one test case fail on this check.

Reply:
Removed. We do `instantiateContext` right after we boot the container.